app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -1108,31 +1166,144 @@ export function testsForRpcMethodSupportingBlockParam(
               // The block-ref middleware will make the request as specified
               // except that the block param is replaced with the latest block
               // number.
-              comms.mockInfuraRpcCall({
+              comms.mockRpcCall({
                 request: buildRequestWithReplacedBlockParam(
                   request,
                   blockParamIndex,
                   '0x100',
                 ),
-                error: `${errorMessagePrefix}: Some message`,
+                response: {
+                  error: 'Some error',
+                  httpStatus,
+                },
                 times: 5,
               });
-              const promiseForResult = withInfuraClient(
+              const promiseForResult = withClient(
+                { type: providerType },
                 async ({ makeRpcCall, clock }) => {
                   return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
                     makeRpcCall(request),
                     clock,
                   );
                 },
               );
+              const err =
+                providerType === 'infura'
+                  ? infuraClientErrorTimeout
+                  : jsonRpcClientErrorTimeout(method);
+              await expect(promiseForResult).rejects.toThrow(err);
+            },
+          );
+        });
+      });
 
-              await expect(promiseForResult).rejects.toThrow(
-                new RegExp(
-                  `^InfuraProvider - cannot complete request\\. All retries exhausted\\..+${errorMessagePrefix}: Some message`,
-                  'su',
-                ),
+      ['ETIMEDOUT', 'ECONNRESET', 'SyntaxError'].forEach(
+        (errorMessagePrefix) => {
+          if (providerType === 'infura' || errorMessagePrefix === 'ETIMEDOUT') {
+            it(`retries the request to Infura up to 5 times if an "${errorMessagePrefix}" error is thrown while making the request, returning the successful result if there is one on the 5th try`, async () => {
```

Comment:
Hmm, this test description has "Infura" in it, but it can be run with the non-Infura client if the current iteration is for `ETIMEDOUT`